src/gc/gc.d

```diff
@@ -2033,6 +2208,13 @@ struct Gcx
         return 1;
     }
 
+    static struct ScanRange
+    {
+        void* pbot;
+        void* ptop;
+        Pool* pool;
```

Comment:
Or you might try to pack the struct and change passing and return to Range + Pool*, using ref return for pop.